### DISCUSSION TITLE
[datasource] add MySQL datasource

### DISCUSSION THREAD

**Discussion by bohutang** [databendlabs/databend#8507] [Ideas]
**Summary**

Add MySQL external datasource.

**Comment by leiysky** [comment]
Does datafuse have mechanism like [FDW(Foreign Data Wrapper)](https://www.postgresql.org/docs/current/postgres-fdw.html)?

It's necessary to support registering external data sources in catalog, with their specific attributes(e.g. host, port, url).

**Comment by bohutang** [comment]
> Does datafuse have mechanism like [FDW(Foreign Data Wrapper)](https://www.postgresql.org/docs/current/postgres-fdw.html)?
> 
> It's necessary to support registering external data sources in catalog, with their specific attributes(e.g. host, port, url).

We are planning to refcator the datasource module to support more datasources.
Such as MySQL, it will be like:
```
ENGINE = MySQL('host:port', 'database', 'table', 'user', 'password')
```
And the ENGINE uri is stored in metastore, when we use one table we first to check it's uri from the metastore and select the right datasource(here is mysql datasource) factory.
BTW, @dantengsky planning working on it.

**Comment by Xuanwo** [comment]
No action to take for now. Let's move to discussion.
